iib/web/models.py

```diff
@@ -306,51 +356,45 @@ def _state_to_json(state):
 
         return rv
 
-    @classmethod
-    def from_json(cls, kwargs):
+    @staticmethod
+    def _from_json(
+        request_kwargs, additional_required_params=None, additional_optional_params=None
+    ):
+        """
+        Validate and process request agnostic parameters
+
+        :param dict request_kwargs: copy of args provided in API request
+        """
         # Validate all required parameters are present
-        required_params = {'bundles', 'binary_image'}
-        optional_params = {'from_index', 'add_arches', 'cnr_token'}
-        missing_params = required_params - kwargs.keys() - optional_params
+        required_params = {'binary_image'} | set(additional_required_params or [])
+        optional_params = {'add_arches', 'cnr_token'} | set(additional_optional_params or [])
+
+        missing_params = required_params - request_kwargs.keys() - optional_params
```

Comment:
- optional_params doens't make sense here optionals does not intersect with required

Reply:
Agreed